src/pdm/models/session.py

```diff
@@ -134,15 +134,18 @@ def __init__(self, *, sources: list[RepositoryConfig], cache_dir: Path, **kwargs
         self._trusted_host_ports: set[tuple[str, int | None]] = set()
         transport: httpx.BaseTransport | None = None
         for s in sources:
+            url = None
+            if s.url is not None:
+                url = httpx.URL(s.url)
+                if s.verify_ssl is False:
+                    self._trusted_host_ports.add((url.host, url.port))
```

Comment:
I have a trouble understanding this fix. This moves the unconditionally adding trust host to a conditional one, then how can I fix the problem?

Reply:
Sorry for making this unclear. I ended up making 2 slightly unrelated changes here.

The first is to move the logic to add a source's URL to `self._trusted_host_ports` to **before** we check for the `pypi` name. This is the part that fixes my issue.

The second is that we add the check for `verify_ssl is False` before we actually add to `self._trusted_host_ports`, which seems to be the behavior for `pdm.utils.get_trusted_hosts` which this code was using prior to `2.13.3`. This check doesn't affect my use case since I explicitly set `verify_ssl = false` in my `pyproject.toml` config for my source. I added it because I thought it was a bit strange to trust sources unconditionally. However, I'm definitely not an expert in this codebase or the best practices around this sort of thing so I'm definitely open to removing this check if you think it's not necessary.

Reply:
thanks, that explains